### ISSUE TITLE
"Null pointer exception may occur here" is never shown

### ISSUE THREAD

**Issue by paulloft** [kalessil/phpinspectionsea#2017]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), version 2026.4                                 |
| Language level | Reproduced on PHP 8.4 and PHP 7.4 (two different, unrelated projects)         |

Current behaviour
--
The "Null pointer exception may occur here" inspection never fires, even for an obvious nullable dereference such as:

```php
function test(?\DateTime $date = null): int
{
    return $date->getTimestamp(); // $date can be null, but no warning is shown
}
```

Other EA Ultimate inspections in the same files/projects continue to work normally, so the plugin as a whole is loaded and active — this appears to be specific to the null-check inspection rather than a full plugin failure.

Toggling the plugin off/on in Settings → Plugins, invalidating caches (File → Invalidate Caches / Restart), and restarting the IDE multiple times does not change anything — the null-check inspection still never triggers. Confirmed programmatically that zero problems are reported for the affected line at any severity down to INFORMATION.

Reproduced in two unrelated projects with different PHP language levels (8.4 and 7.4), so this doesn't look project- or PHP-version-specific.

Separately, the following exception is logged on every IDE startup — not sure if it's related, but logging it in case it's relevant:

```
com.intellij.diagnostic.PluginException: Migrate com.kalessil.phpStorm.phpInspectionsEA.EAUltimateSidebarComponent to ProjectActivity [Plugin: com.kalessil.phpStorm.phpInspectionsUltimate-mp]
at com.intellij.ide.startup.impl.StartupManagerImpl.doRunPostStartupActivities(StartupManagerImpl.kt:340)
at com.intellij.ide.startup.impl.StartupManagerImpl.access$doRunPostStartupActivities(StartupManagerImpl.kt:95)
at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$3$2.invokeSuspend(StartupManagerImpl.kt:233)
at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$3$2.invoke(StartupManagerImpl.kt)
at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$3$2.invoke(StartupManagerImpl.kt)
at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:67)
at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)
at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:165)
at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$3.invokeSuspend(StartupManagerImpl.kt:232)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:610)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runDefaultDispatcherTask(CoroutineScheduler.kt:1194)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:906)
at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:762)
```
Expected behaviour
--
The "Null pointer exception may occur here" inspection should fire normally for nullable dereferences, consistent with the rest of the plugin's inspections which work fine.

Environment details
---
- No other PHP static-analysis plugin installed alongside it (single PHP inspections plugin: EA Ultimate — ruled out the known short-name collision with EA Extended).
- Plugin was enabled the whole time in Settings → Plugins; toggling it off/on had no effect.

Information from Help - About dialog
---
IDE: PhpStorm (build #PS-261.26222.71, Thu, 2 Jul 2026 11:27:00 GMT)
Runtime version: 25.0.3+9-b508.16 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
OS: macOS (26.5.2)
Plugin: Php Inspections (EA Ultimate) 2026.4

**Comment by paulloft** [comment]
Additional finding: I tested with Php Inspections (EA Ultimate) 2025.5, and the "Null pointer exception may occur here" inspection works correctly there.

So this looks like a regression introduced between plugin version 2025.5 and 2026.4:
- Plugin 2026.4 — inspection does NOT fire.
- Plugin 2025.5 — inspection works correctly.

Everything else (project, code sample, PHP language level) was identical between the two tests.

**Comment by pdjshog** [comment]
same problem

**Comment by ea-inspections-team** [comment]
Thank you for reporting @paulloft @pdjshog!

- The inspection has been retired, but I am re-enabling it for Php Inspections (EA Ultimate) as `Null reference (legacy, unsupported)`
- The diagnostic exception on startup is something new - I'll look into it for the next release
